### PR TITLE
refactor: update java imports in `Environment.flix`

### PR DESCRIPTION
**Pull request by jaschdoc**
Related to https://github.com/flix/flix/issues/7872

### CODE REVIEW

File: main/src/library/Environment.flix [hunk 1]
```diff
@@ -28,15 +31,15 @@ mod Environment {
...
+        import static java.lang.System.getenv(): JMap \ rc;
+        import java.util.Map.entrySet(): JSet \ { rc };
+        import java.util.Set.iterator(): JIterator \ { rc };
         let _ = rc; // Avoids redundancy error.
         try {
             let iter = getenv() |> entrySet |> iterator;
             getEnvHelper(iter, Map.empty())
         } catch {
-            case _:##java.lang.Exception => Map.empty()
+            case _: ##java.lang.Exception => Map.empty()
```
Comment (jaschdoc) [inline main/src/library/Environment.flix:42]: The try-catch rule just looks up the java class in `Resolver` and does not consider imports. Should I open a ticket, or is this going to be fixed?

Comment (jaschdoc) [reply main/src/library/Environment.flix:42]: The solution is just to first do a lookup in the imports to get the actual java class and if that fails then fall back to trying to find a class for the literal (which is the current impl). Maybe the other way around?

Comment (jaschdoc) [reply main/src/library/Environment.flix:42]: It seems this should be fixed in namer or maybe weeder even?

Comment (magnus-madsen) [reply main/src/library/Environment.flix:42]: @mlutze Any insights?

@jaschdoc If you can fix it, probably in Resolver, you are welcome.

Comment (jaschdoc) [reply main/src/library/Environment.flix:43]: I'll take a stab at it in a new pr

File: main/src/library/Environment.flix [hunk 2]
```diff
@@ -126,15 +130,15 @@ mod Environment {
...
     def getEnvHelper(iter: ##java.util.Iterator, m: Map[String, String]): Map[String, String] =
         import java.util.Iterator.hasNext(): Bool \ {};
-        import java.util.Iterator.next(): ##java.lang.Object \ {};
-        import java.util.Map$Entry.getKey(): ##java.lang.Object \ {};
-        import java.util.Map$Entry.getValue(): ##java.lang.Object \ {};
+        import java.util.Iterator.next(): Object \ {};
+        import java.util.Map$Entry.getKey(): Object \ {};
+        import java.util.Map$Entry.getValue(): Object \ {};
         if (hasNext(iter))
             let e = unchecked_cast(next(iter) as ##java.util.Map$Entry);
```
Comment (jaschdoc) [inline main/src/library/Environment.flix:143]: Same thing applies to `##java.util.Map$Entry`

File: main/src/library/Environment.flix (hunk 1 above)
Comment (mlutze) [reply main/src/library/Environment.flix:43]: I would guess that this is parsed specifically as a Java type, so you'll might need changes from parser to resolver.

**Review by magnus-madsen — APPROVED** [review]

**Pull request merged**